openedx/core/djangoapps/oauth_dispatch/docs/decisions/0006-enforce-scopes-in-LMS-APIs.rst

```diff
@@ -0,0 +1,225 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+(**Note to MS:** This seems to be the DOT recommended way of adding 
+available scopes support.)
+
+* In order to allow open edX operators to a priori limit the
+  types of access an Application can request, we will allow them
+  to configure Application-specific "available scopes".
+
+* Introduce a new data model that associates available scopes with
+  DOT Applications.
+
+* Introduce a new Scopes Backend that extends DOT's SettingsScopes_
+  backend and overrides the implementation of get_available_scopes_.
+
+* The new backend will query the new data model to retrieve
+  available scopes.
+
+.. _get_available_scopes: https://github.com/evonove/django-oauth-toolkit/blob/2129f32f55cda950ef220c130dc7de55bea29caf/oauth2_provider/scopes.py#L17
+.. _SettingsScopes: https://github.com/evonove/django-oauth-toolkit/blob/2129f32f55cda950ef220c130dc7de55bea29caf/oauth2_provider/scopes.py#L39
+
+4. Associate Available Organizations with Applications
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+(**Note to MS:**  Some details here are different from the implementation
+in your PR. See further explanations in the Consequences section below.)
+
+* In order to allow Applications to access data for their own
+  organization without inadvertently or maliciously gaining access
+  to data for other organizations, Applications need to be
+  linked to their own organization. To support this, open edX 
+  operators can configure Application-specific "available organizations",
+  which are akin to Application-specific "available scopes".
+
+* Introduce a new data model that associates available organizations
+  with DOT Applications.
+
+* The new data model will have a Foreign Key to the Organization_ table.
+  It will essentially be a many-to-many relationship between Organizations
+  and DOT Applications.
+
+* The organization associated with the Application will be included
+  in the JWT tokens requested by the Application.
+
+  * JwtBuilder_'s *build_token* functionality will be extended to include
+    the organization value in the token's payload. This payload is
+    cryptographically signed and so binds and limits the scopes in the
+    token to the organization.
+
+  * Since the organization value is in the token, any relying parties
+    that receive the token (including microservices) will be able to
+    enforce the scopes as limited to the organization.
+
+* **Question:** Should we distinguish between user_org and course_org?  
+  The former is the Enterprise relationship while the latter is the
+  content provider relationship.  In the future, we will also have
+  sponsorship relationship. This depends on whether large organizations
+  will want to compartmentalize their Applications.
+
+.. _Organization: https://github.com/edx/edx-organizations/blob/fa137881be9b7d330062bc32655a00c68635cfed/organizations/models.py#L14
+
+5. Introduce a new Permission class to enforce scopes
```

Comment:
Reminder: Address non-DRF endpoints
- See how we can protect non-DRF endpoints. - A decorator?
- See which ones exist today

Reply:
done.